dist/index.d.ts

```diff
@@ -1,8 +1,10 @@
 declare const makeErrorChecker: (e: object) => {
+    [x: string]: boolean | object;
```

Comment:
Ding! Common TS mistake. This shouldn't be here. Basically, you're saying: _this object can have any keys in it_, which kinda defeats the whole purpose of specifying the keys just after it.